Xdl.Internship.Core/Xdl.Internship.Core.DataAccess.MongoDB/Repositories/IMongoRepository.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using Xdl.Internship.Core.Models.MongoDB;
+
+namespace Xdl.Internship.Core.DataAccess.MongoDB.Repositories
+{
+    public interface IMongoRepository<TDocument>
+        where TDocument : IModelBase
+    {
+        Task<ICollection<TDocument>> GetAsync(ObjectId id, CancellationToken cancellationToken = default);
+
+        Task<ICollection<TDocument>> GetAllAsync(CancellationToken cancellationToken = default);
```

Comment:
FindAllAsync, if you use FindOneAsync naming

Reply:
resolved